src/Lang/Cpp/CppDocGen.ts

```diff
@@ -105,6 +105,18 @@ export class CppDocGen implements IDocGen {
         return template.replace(replace, param);
     }
 
+    protected getMultiTemplatedString(replace: string[], template: string, param: string[]): string {
+        // FIXME I find this argument order a bit strange.  I would probably have template first
+        // For each replace entry, attempt to replace it with the corresponding param in the template
+        for (let i = 0; i < replace.length; i++) {
+            if (i < param.length) {
+              template = template.replace(replace[i], param[i]);
+            }
+            // TODO: warn if mismatch string lengths?  Probably should use tuple of tuples
```

Comment:
What do you mean by that?

Reply:
I'm just a bit wary of passing two different sized arrays to a function as different arguments.  A single argument with an array of [string,string] tuples might be safer (or at least caught at compile time?)

Reply:
Yep. Good thinking!